floyd/cli/utils.py

```diff
@@ -55,28 +60,139 @@ def get_data_id(data_str):
         return data_str
 
 
-def normalize_data_name(data_name):
-    if data_name.endswith('/output'):
-        name_parts = data_name.split('/')
-        if len(name_parts) <= 4:
-            name_parts.insert(1, 'projects')
-            data_name = '/'.join(name_parts)
-        return data_name
+def normalize_data_name(raw_name, default_username=None, default_dataset_name=None):
+    if raw_name.endswith('/output'):
+        return normalize_job_name(raw_name[:-len('/output')], default_username, default_dataset_name) + '/output'
+
+    name_parts = raw_name.split('/')
+
+    username = default_username or current_username()
+    name = default_dataset_name or current_experiment_name()
+    number = None  # current version number
+
+    # When nothing is passed, use all the defaults
+    if not raw_name:
+        pass
+    elif len(name_parts) == 4:
+        # mckay/datasets/foo/1
+        username, _, name, number = name_parts
+    elif len(name_parts) == 3:
+
+        if name_parts[2].isdigit():
+            # mckay/foo/1
+            username, name, number = name_parts
+        else:
+            # mckay/projects/foo
+            username, _, name = name_parts
+    elif len(name_parts) == 2:
+        if name_parts[1].isdigit():
+            # foo/1
+            name, number = name_parts
+        else:
+            # mckay/foo
+            username, name = name_parts
+    elif len(name_parts) == 1:
+        if name_parts[0].isdigit():
+            # 1
+            number = name_parts[0]
+        else:
+            # foo
+            name = name_parts[0]
     else:
-        name_parts = data_name.split('/')
-        if len(name_parts) <= 3:
-            name_parts.insert(1, 'datasets')
-            data_name = '/'.join(name_parts)
-        return data_name
+        return raw_name
+
+    # If no number/version is found, query the API for the most recent version
+    if number is None:
+        name_from_api = get_lateset_dataset_version(username, name)
```

Comment:
@houqp Can we just support a "latest" version on the server? Then the cli can just use that instead of querying for the latest version.

Reply:
It's already supported, if you mount data without version like `namespace/datasets/name`, it will default to latest version.